Adapter/Algolia.php

```diff
@@ -90,140 +96,200 @@ public function __construct(
      *
      * @return array
      */
-    private function getAggregations($request, $table, $documents)
+    private function getContextParams($storeId)
     {
-        if (version_compare($this->config->getMagentoVersion(), '2.1.0', '<') === true) {
-            return $this->aggregationBuilder->build($request, $table);
-        }
+        $contextParams = [];
+        $category = $this->registry->registry('current_category');
+        $page = !is_null($this->request->getParam('page')) ?
+            (int) $this->request->getParam('page') - 1 :
+            0;
+
+        if ($category) {
+            $contextParams = [
+                'facetFilters' => [
+                    'categoryIds:' . $category->getEntityId(),
+                ],
+                'page' => $page,
+            ];
+
+            $facetFilters = [];
+
+            foreach ($this->config->getFacets($storeId) as $facet) {
+                if (!is_null($this->request->getParam($facet['attribute']))) {
+                    if ($facet['type'] === 'conjunctive') {
+                        $facetValues = explode('~', $this->request->getParam($facet['attribute']));
+                        foreach ($facetValues as $key => $facetValue) {
+                            $facetFilters[] = $facet['attribute'] . ':' . $facetValue;
+                        }
+                    }
+
+                    if ($facet['type'] === 'disjunctive') {
+                        $facetValues = explode('~', $this->request->getParam($facet['attribute']));
+                        if (count($facetValues) > 1) {
+                            foreach ($facetValues as $key => $facetValue) {
+                                $facetValues[$key] = $facet['attribute'] . ':' . $facetValue;
+                            }
+                            $facetFilters[] = $facetValues;
+                        }
+                        if (count($facetValues) == 1) {
+                            $facetFilters[] = $facet['attribute'] . ':' . $facetValues[0];
+                        }
+                    }
+                }
+            }
 
-        return $this->aggregationBuilder->build($request, $table, $documents);
-    }
+            $contextParams['facetFilters'] = array_merge($contextParams['facetFilters'], $facetFilters);
 
-    /** @return string */
-    private function getGetDocumentMethod()
-    {
-        if (version_compare($this->config->getMagentoVersion(), '2.1.0', '<') === true) {
-            return 'getDocument20';
+            // Handle price filtering
+            $currencyCode = $this->storeManager->getStore()->getCurrentCurrencyCode();
+            $priceSliders[] = 'price_' . $currencyCode . '_default';
+
+            if ($this->config->isCustomerGroupsEnabled($storeId)) {
+                $groupCollection = $this->objectManager
+                    ->create('Magento\Customer\Model\ResourceModel\Group\Collection');
+
+                foreach ($groupCollection as $group) {
```

Comment:
The query doesn't have to iterate over all customer groups. You can get a customer group of the currently logged in customer and filter only on his specific price attribute, no?